x/genutil/client/cli/init.go

```diff
@@ -158,7 +162,16 @@ func InitCmd(mm *module.Manager) *cobra.Command {
 			appGenesis.InitialHeight = initHeight
 			appGenesis.Consensus = &types.ConsensusGenesis{
 				Validators: nil,
+				Params: &cmttypes.ConsensusParams{
+					Validator: cmttypes.ValidatorParams{},
+				},
+			}
+
+			consensusKey, err := cmd.Flags().GetString(FlagConsensusKeyAlgo)
+			if err != nil {
+				return errorsmod.Wrap(err, "Failed to get consensus key algo")
 			}
+			appGenesis.Consensus.Params.Validator.PubKeyTypes = []string{consensusKey}
```

Comment:
the consensus key is validated in the `ExportGenesisFile` step. It would return `params.Validator.PubKeyTypes[%d], %s, is an unknown pubkey type` for unsupported types. It is ok for me to rely on this and not fail fast